### PR TITLE
OKTA-543909 - Remove keys check when adding EnrollingFactors preventing TOTP methods to be added

### PR DESCRIPTION
**Pull request by moisesolmedo-okta**
### Problem Analysis (Technical)
OV was broken due to recent MyAccount refactor. TOTP method was not being added to the list of Methods due to unnecessarily checking for the factor.keys on the totp enrollment and therefore OV was not showing any enrollment after adding an account.

### Solution (Technical)
Remove extra check for factor.keys since all Methods add this property since it's optional.

### Steps to reproduce:

Actual result:

Expected result:

### CODE REVIEW

File: Sources/DeviceAuthenticator/Networking/LegacyServerAPI.swift [hunk 1]
```diff
@@ -211,24 +211,20 @@ class LegacyServerAPI: ServerAPIProtocol {
...
-                                                                                       apsEnvironment: factor.methodType == .push ? factor.apsEnvironment : nil,
-                                                                                       supportUserVerification: factor.supportUserVerification,
-                                                                                       isFipsCompliant: factor.isFipsCompliant,
-                                                                                       keys: factor.keys,
-                                                                                       capabilities: capabilities)
-
-                return methodModel
-            } else {
-                return nil
+            var transactionTypes: [MethodSettingsModel.TransactionType] = [.LOGIN]
```
Comment (IldarAbdullin-okta) [inline Sources/DeviceAuthenticator/Networking/LegacyServerAPI.swift:214]: Do you want to set this only for push, since it is relevant only for push?

**Review by IldarAbdullin-okta — APPROVED** [review]
LGTM!

File: Sources/DeviceAuthenticator/Networking/LegacyServerAPI.swift (hunk 1 above)
Comment (moisesolmedo-okta) [reply Sources/DeviceAuthenticator/Networking/LegacyServerAPI.swift]: good point

**Review by IldarAbdullin-okta — APPROVED** [review]
👍

**Pull request merged**